olclient/entity_helpers/work.py

```diff
@@ -0,0 +1,172 @@
+from __future__ import annotations
+
+import json
+import logging
+import re
+from typing import List, Dict, Optional, Any
+
+import backoff
+
+from olclient.common import Entity, Book
+from olclient.helper_classes.results import Results
+from olclient.utils import merge_unique_lists, get_text_value
+
+logger = logging.getLogger('open_library_work')
+
+
+def get_work_helper_class(ol_context):
+    class Work(Entity):
+
+        OL = ol_context
+
+        def __init__(self, olid, identifiers=None, **kwargs):
+            super().__init__(identifiers)
+            self.olid = olid
+            self._editions = []
+            self.description = get_text_value(kwargs.pop('description', None))
+            self.notes = get_text_value(kwargs.pop('notes', None))
+            for kwarg in kwargs:
+                setattr(self, kwarg, kwargs[kwarg])
+
+        def json(self) -> dict:
+            """Returns a dict JSON representation of an OL Work suitable
+            for saving back to Open Library via its APIs.
+            """
+            exclude = ['_editions', 'olid']
+            data = {k: v for k, v in self.__dict__.items() if v and k not in exclude}
+            data['key'] = '/works/' + self.olid
+            data['type'] = {'key': '/type/work'}
+            if data.get('description'):
+                data['description'] = {
+                    'type': '/type/text',
+                    'value': data['description'],
+                }
+            if data.get('notes'):
+                data['notes'] = {'type': '/type/text', 'value': data['notes']}
+            return data
+
+        def validate(self) -> None:
+            return self.OL.validate(self, 'work.schema.json')
+
+        @property
+        def editions(self):
+            url = f'{self.OL.base_url}/works/{self.olid}/editions.json'
+            try:
+                r_json: Dict[Any, Any] = self.OL.session.get(url).json()
+                editions: List[Any] = r_json.get('entries', [])
+                while True:
+                    next_page_link: Optional[str] = r_json.get('links', {}).get('next')
+                    if next_page_link is not None:
+                        r_json: Dict[Any, Any] = self.OL.session.get(
+                            self.OL.base_url + next_page_link
+                        ).json()
+                        editions.extend(r_json.get('entries', []))
+                    else:
+                        break
+            except BaseException:
+                return []
+
+            self._editions = [
+                self.OL.Edition(**self.OL.Edition.ol_edition_json_to_book_args(ed))
+                for ed in editions
+            ]
+            return self._editions
+
+        @classmethod
+        def create(cls, book: Book, debug=False) -> Work:
+            try:
+                book.publish_date = re.findall(r'[\d]{4}', book.publish_date)[0]
+            except:
```

Comment:
https://realpython.com/the-most-diabolical-python-antipattern/

Reply:
You are right. This is not a good idea to have blanket except clauses which hide all errors. 

It was a mistake from my part to not mention context in the description 😅 , added description to the PR now. The intent of the PR is just to move a part of the code from `openlibrary.py` to another file and resolve all the circular dependency issues and see if we can progressively do this. 

All this code is what was already in the codebase, just moved into a different file. If you see this as a chance to refactor/cleanup pieces of code, I would agree with that too. 

Should we do a refactor/cleanup in the same PR? If yes, then I'll go ahead and make changes.